sdk/python/foundation-models/system/content-safety/deploy_with_content_safety.ipynb

```diff
@@ -0,0 +1,1238 @@
+    "### This notebook will walk you through the steps to create an __Azure AI Content Safety__ enabled __LLaMA__ online/batch endpoint.\n",
+    "### The steps are:\n",
+    "1. Create an __Azure AI Content Safety__ resource for moderating the request from user and response from the __LLaMA__ model.\n",
+    "2. Create a new __Azure AI Content Safety__ enabled __LLaMA__ online endpoint with a custom score.py which will integrate with the __Azure AI Content Safety__ resource to moderate the response from the __LLaMA__ model and the request from the user, but to make the custom score.py to sucessfully autheticate to the __Azure AI Content Safety__ resource, we have 2 options:\n",
+    "    1. __UAI__, recommended but more complex approach, is to create a __User Assigned Identity (UAI)__ and assign appropriate roles to the __UAI__. Then, the custom score.py can obtain the access token of the __UAI__ from the AAD server to access the Azure AI Content Safety resource.\n",
+    "    2. __Environment variable__, simpler but less secure approach, is to just pass the access key of the __Azure AI Content Safety__ resource to the custom score.py via environment variable, then the custom score.py can use the key directly to access the Azure AI Content Safety resource, this option is less secure than the first option, if someone in your org has access to the endpoint, he/she can get the access key from the environment variable and use it to access the Azure AI Content Safety resource.\n",
+    "  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Please update below variable to EnvVar if you want to use the Environment variable approach."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "aacs_authentication_mode = \"UAI\"  # UAI or EnvVar, if EnvVar is used, the access keys of AACS and LLaMA online endpoint will be set to environment variables."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Online endpoint or Batch endpoint\n",
+    "\n",
+    "Azure Machine Learning allows you to deploy models using online endpoints and batch endpoints. Online endpoints are designed for real-time inference—when you invoke the endpoint, the results are returned in the endpoint's response. Batch endpoints, on the other hand, are designed for long-running batch inference. Each time you invoke a batch endpoint you generate a batch job that performs the actual work. \n",
+    "\n",
+    "For more details, please refer to [Online or Batch endpoint](https://learn.microsoft.com/en-us/azure/machine-learning/concept-endpoints?view=azureml-api-2#when-to-use-online-vs-batch-endpoint-for-your-use-case).\n",
+    "\n",
+    "Online endpoint is enabled by default in this notebook, if you want to create a batch endpoint, please update the following variable to __batch__\n",
```

Comment:
I think for batch the scoring script also needs to be modified e.g. https://github.com/Azure/azureml-examples/blob/main/cli/endpoints/batch/deploy-models/custom-outputs-parquet/code/batch_driver.py , so you might want to add a separate scoring script for batch altogether.